### DISCUSSION TITLE
`docker start` VS `docker compose up` behavior after a container was stopped with `docker stop`

### DISCUSSION THREAD

**Discussion by simon-liebehenschel** [moby/moby#45471] [Q&A] [answered]
What is the difference between `docker start` VS `docker coppose up`? Which one should I use and in which cases?

`docker compose up -d` + `docker stop <container_name>` + `docker compose up -d`

VS

`docker compose up -d` + `docker stop <container_name>` + `docker start <container_name>`

---

I am asking because `docker compose up` will remove files outside of mounted volumes, but `docker start <container_name>` persist all files even outside of volumes.  What exactly is done under the hood?

**Comment by laurazard** [comment, marked as answer]
Hiya @AIGeneratedUsername – when you create/run a container, a filesystem is created for that container. This filesystem is ephemeral, and is removed when that container is deleted, so any changes made during the container's execution will be lost (unless these are made to a persisted storage, like a mounted volume).

`docker stop` stops a container, but does not delete it, so filesystem changes are not lost. Running `docker start` afterwards starts the container again, with the same filesystem, so any changes still exist there.

However, `docker compose up` employs a reconciliation logic, and might do a number of things when you run it – it might just start your previous containers, but if the Compose file was changed in the meantime or some other aspect of it's configuration, it will recreate the container, deleting the previous containers filesystem and losing any changes made not on a mounted volume.

You can run `docker compose start` if you just want Compose to start stopped containers. You can also read more about what `docker compose up` does [here](https://docs.docker.com/engine/reference/commandline/compose_up/). Specifically, this part might interest you:

```yaml
If there are existing containers for a service, and the service’s configuration or image was changed
after the container’s creation, docker compose up picks up the changes by stopping and recreating
the containers (preserving mounted volumes). To prevent Compose from picking up changes, use
the --no-recreate flag.
```
